.github/scripts/test_open_data_assets.py

```diff
@@ -0,0 +1,212 @@
+ASSET_API_QUERY_PARAMS = {"$query": "SELECT COUNT(*),year GROUP BY year"}
+DBT = dbtRunner()
+
+# Data for the most recent two years are permitted to be slightly different,
+# according to this buffer value
+BUFFER = 0.02
+
+
+def main() -> None:
+    """Entrypoint for the script."""
+    dbt_manifest_path = sys.argv[1]
+    with open(dbt_manifest_path) as dbt_manifest_file:
+        dbt_manifest_json = json.loads(dbt_manifest_file.read())
+
+    diffs: typing.List[typing.List[typing.Dict]] = []
+
+    for exposure in dbt_manifest_json["exposures"].values():
+        asset_name = exposure["label"]
+        asset_url = exposure["url"]
+        asset_id = asset_url.split("/")[-1]
+
+        # For now, assume just one dependency table, since all of our open
+        # data assets currently have a 1:1 relationship with a view.
+        fq_source_model_name = exposure["depends_on"]["nodes"][0]
+
+        # Dependency tables in the DAG are fully qualified, so we have to
+        # strip the prefix to get the schema and tablename.
+        source_model_name = ".".join(fq_source_model_name.split(".")[2:])
+
+        print(f"Comparing asset '{asset_name}' to model '{source_model_name}'")
+
+        asset_api_url = ASSET_API_URL.format(asset_id=asset_id)
+        response = requests.get(asset_api_url, params=ASSET_API_QUERY_PARAMS)
+        if not response.ok:
+            print("Encountered error in request to asset endpoint")
+            response.raise_for_status()
+
+        asset_row_counts_by_year = response.json()
+
+        dbt_output = io.StringIO()
+        with contextlib.redirect_stdout(dbt_output):
+            dbt_result = DBT.invoke(
+                [
+                    "--quiet",
+                    "run-operation",
+                    "row_count_by_year",
+                    "--args",
+                    f"{{model: '{source_model_name}', print: true}}",
+                    "--target",
+                    "prod",
+                ]
+            )
+
+        if not dbt_result.success:
+            print("Encountered error in dbt call")
+            raise ValueError(dbt_result.exception)
+
+        source_model_row_counts_by_year = json.loads(dbt_output.getvalue())
+
+        if diff := diff_row_counts(
+            source_model_name,
+            source_model_row_counts_by_year,
+            asset_name,
+            asset_row_counts_by_year,
+        ):
+            diffs.append(diff)
+
+    if diffs:
+        print()
+        print(
+            "The following view/asset pairs had mismatching row counts "
+            "by year:"
+        )
+        for diff in diffs:
+            print()
+            diff_table = agate.Table.from_object(diff)
+            # Note that agate adds thousands separators to years, even when
+            # they're strings (as in our data). This is super annoying but
+            # it's still the fastest way we know to print a clean table
+            # in Python :\
```

Comment:
Any ideas about better ways to do this? Agate is a dependency of dbt so it was the easy option, but I'm sure there's another way!